### PR TITLE
Make PyTorch experiments deterministic

### PR DESCRIPTION
**Pull request by utkarshjp7**
This PR makes makes PyTorch experiments deterministic.

It follows these instructions -> https://pytorch.org/docs/stable/notes/randomness.html

### CODE REVIEW

File: src/garage/experiment/deterministic.py [hunk 1]
```diff
@@ -2,26 +2,36 @@
...
 import sys
 
+from dowel import logger
 import numpy as np
 
 seed_ = None
 
 
-def set_seed(seed):
+def set_seed(seed, pytorch=False):
```
Comment (ryanjulian) [inline src/garage/experiment/deterministic.py]: i'm not in favor of this option. if you want your experiment to be deterministic, you set the seed.

File: tests/garage/experiment/test_deterministic.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import torch
+
+from garage.experiment import deterministic
+
+
+def test_deterministic_pytorch():
```
Comment (ryanjulian) [inline tests/garage/experiment/test_deterministic.py]: omg nice. can you add one for TF while you're here?

Comment (ryanjulian) [reply tests/garage/experiment/test_deterministic.py]: and numpy and random.random, as well?

File: src/garage/experiment/deterministic.py (hunk 1 above)
Comment (utkarshjp7) [reply src/garage/experiment/deterministic.py]: I added this because making PyTorch deterministic has performance impact, but on second thought, it doesn't really make sense since users won't be running both PyTorch and Tensorflow experiments together. I will remove this.

Comment (ryanjulian) [reply src/garage/experiment/deterministic.py]: you can print a warning with `warnings.warn` that this has a negative impact on performance.

BTW, i thought it only impacts performance on GPU, right?

File: src/garage/experiment/deterministic.py [hunk 3]
```diff
@@ -15,13 +16,21 @@ def set_seed(seed):
...
     global seed_
     seed_ = seed
     random.seed(seed)
     np.random.seed(seed)
     if 'tensorflow' in sys.modules:
-        import tensorflow as tf
+        import tensorflow as tf  # pylint: disable=import-outside-toplevel
         tf.compat.v1.set_random_seed(seed)
+    if 'torch' in sys.modules:
+        logger.log('Enabeling deterministic mode in pytorch mode can have a \
```
Comment (ryanjulian) [inline src/garage/experiment/deterministic.py]: `warnings.warn` is a better way for this right now. the logger might not yet have a StdOutput

Comment (ryanjulian) [reply src/garage/experiment/deterministic.py]: also, if this only affects GPU, please say so.

File: tests/garage/experiment/test_deterministic.py (hunk 2 above)
Comment (utkarshjp7) [reply tests/garage/experiment/test_deterministic.py]: Yes, I added them as well.

File: src/garage/experiment/deterministic.py (hunk 1 above)
Comment (utkarshjp7) [reply src/garage/experiment/deterministic.py]: yes, you are right. I changed the text.

File: src/garage/experiment/deterministic.py (hunk 3 above)
Comment (utkarshjp7) [reply src/garage/experiment/deterministic.py]: Using `warnings.warn` now.

File: src/garage/experiment/deterministic.py [hunk 4]
```diff
@@ -15,13 +16,22 @@ def set_seed(seed):
...
     seed_ = seed
     random.seed(seed)
     np.random.seed(seed)
     if 'tensorflow' in sys.modules:
-        import tensorflow as tf
+        import tensorflow as tf  # pylint: disable=import-outside-toplevel
         tf.compat.v1.set_random_seed(seed)
+    if 'torch' in sys.modules:
+        warnings.warn(
+            'Enabeling deterministic mode in PyTorch can have a performance \
```
Comment (ryanjulian) [inline src/garage/experiment/deterministic.py]: python joins strings between () implicitly, even if they are on different lines.

e.g.
```python
warnings.warm(
    'foo'
    'bar')
```
will result in warning text `foobar`

**Comment by ryanjulian** [comment]
you can put batch the `ready-to-merge` label when you fix the string splitting

**Comment by utkarshjp7** [comment]
Can this be force pushed? The diff coverage is failing.

**Comment by ryanjulian** [comment]
@Mergifyio rebase

**Comment by ryanjulian** [comment]
Yep. Just ping me when it's green.

**Comment by utkarshjp7** [comment]
> Yep. Just ping me when it's green.

Its green now.

**Comment by utkarshjp7** [comment]
@Mergifyio Rebase

**Comment by utkarshjp7** [comment]
@Mergifyio rebase

**Comment by utkarshjp7** [comment]
@ryanjulian  This is ready to be force-pushed.

**Comment by ryanjulian** [comment]
@Mergifyio backport release-2019.10
